qiskit/chemistry/ground_state_calculation/mes_ground_state_calculation.py

```diff
@@ -0,0 +1,178 @@
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Ground state computation using a minimum eigensolver."""
+
+from typing import Union, List, Any, Optional, Dict
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Instruction
+from qiskit.quantum_info import Statevector
+from qiskit.result import Result
+from qiskit.aqua.algorithms import MinimumEigensolver
+from qiskit.aqua.operators import OperatorBase, WeightedPauliOperator, StateFn, CircuitSampler
+from qiskit.chemistry import FermionicOperator
+from qiskit.chemistry.drivers import BaseDriver
+from qiskit.chemistry.ground_state_calculation import GroundStateCalculation
+from qiskit.chemistry.qubit_transformations import QubitOperatorTransformation
+from qiskit.chemistry.results import EigenstateResult
+
+from .mes_factories import MESFactory
+
+
+class MinimumEigensolverGroundStateCalculation(GroundStateCalculation):
+    """Ground state computation using a minimum eigensolver."""
+
+    def __init__(self, transformation: QubitOperatorTransformation,
+                 solver: Union[MinimumEigensolver, MESFactory]) -> None:
+        """
+
+        Args:
+            transformation: Qubit Operator Transformation
+            solver: Minimum Eigensolver or MESFactory object, e.g. the VQEUCCSDFactory.
+        """
+        super().__init__(transformation)
+        self._solver = solver
+
+    @property
+    def solver(self) -> Union[MinimumEigensolver, MESFactory]:
+        """Returns the minimum eigensolver or factory."""
+        return self._solver
+
+    @solver.setter
+    def solver(self, solver: Union[MinimumEigensolver, MESFactory]) -> None:
+        """Sets the minimum eigensolver or factory."""
+        self._solver = solver
+
+    def returns_groundstate(self) -> bool:
+        """TODO
+        whether the eigensolver returns the ground state or only ground state energy."""
+
+        return False
+
+    def compute_groundstate(self, driver: BaseDriver,
+                            aux_operators: Optional[List[Any]] = None
+                            ) -> EigenstateResult:
+        """Compute Ground State properties.
+
+        Args:
+            driver: a chemistry driver object which defines the chemical problem that is to be
+                    solved by this calculation.
+            aux_operators: Additional auxiliary operators to evaluate at the ground state.
+                Depending on whether a fermionic or bosonic system is solved, the type of the
+                operators must be ``FermionicOperator`` or ``BosonicOperator``, respectively.
```

Comment:
Then make the List a be a Union of these if they can be intermixed i.e. get rid of Any as the type.

Reply:
to be pushed soon